### PR TITLE
Fix warnings and valid errors reported by coverity

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Fix some of the warnings and valid errors reported by coverity.

### CODE REVIEW

File: sound/soc/sof/ipc4-pcm.c [hunk 1]
```diff
@@ -708,6 +708,9 @@ static int sof_ipc4_pcm_hw_params(struct snd_soc_component *component,
 	struct snd_sof_pcm *spcm;
 
 	spcm = snd_sof_find_spcm_dai(component, rtd);
+	if (!spcm)
+		return -ENODEV;
```
Comment (bardliao) [inline sound/soc/sof/ipc4-pcm.c]: Shouldn't it be `-EINVAL`?

File: sound/soc/sof/ipc4-topology.c [hunk 2]
```diff
@@ -1859,12 +1859,11 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	struct sof_ipc4_audio_format *in_audio_fmt;
 	u32 out_ref_rate, out_ref_channels, out_ref_valid_bits;
 	int output_format_index, input_format_index;
-	int ret;
 
 	input_format_index = sof_ipc4_init_input_audio_fmt(sdev, swidget, &src->base_config,
 							   pipeline_params, available_fmt);
 	if (input_format_index < 0)
-		return ret;
+		return input_format_index;
```
Comment (plbossart) [inline sound/soc/sof/ipc4-topology.c:1866]: the commit should be titled 
fixup! ASoC: SOF: ipc4-topology: Fix the output reference params for SRC

see 53afe5ec6789

File: sound/soc/sof/ipc4-pcm.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/ipc4-pcm.c]: yes indeed this should be -EINVAL as done in all other cases

File: sound/soc/intel/skylake/skl-sst-utils.c [hunk 3]
```diff
@@ -311,6 +311,9 @@ int snd_skl_parse_uuids(struct sst_dsp *ctx, const struct firmware *fw,
 
 	return 0;
 
+free_uuid_module:
+	kfree(module);
+
```
Comment (plbossart) [inline sound/soc/intel/skylake/skl-sst-utils.c]: @aiChaoSONG please drop this patch, we don't use/own this driver.

@crojewsk @crojewsk-intel please look into this on your side.

**Review by plbossart — CHANGES_REQUESTED** [review]

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

File: sound/soc/intel/skylake/skl-sst-utils.c (hunk 3 above)
Comment (crojewsk-intel) [reply sound/soc/intel/skylake/skl-sst-utils.c]: @plbossart sure thing, will do. Sorry for the delay, came back from holidays recently and didn't notice the request at first.

**Pull request merged**